### PR TITLE
Use FastWriter for TlvJson

### PR DESCRIPTION
**Pull request by CodeChronos928**
TlvJson now outputs a compact machine-readable string, instead of one that's human-readable.

#### Problem

I'm using TlvJson to generate attribute values in JSON format, and I want the string to be compact.  It's currently outputting human-readable JSON that's full of extra white space.

I don't believe this library was meant for debug purposes only, so I propose to change the TlvJson behavior rather than re-formatting the JSON on the app side.  Folks that are using it to debug can always run it through `python -m json.tool`

#### Change overview

Adjust TlvJson to output a compact machine-readable JSON format.

##### Before

```
{
   "value" : {
      "0" : 20,
      "1" : true,
      "2" : 0,
      "3" : "AQIDBP/+mYjdzQ==",
      "4" : "hello",
      "5" : 0,
      "6" : 1.0,
      "7" : 1.0
   }
}
```

##### After

```
{"value":{"0":20,"1":true,"2":0,"3":"AQIDBP/+mYjdzQ==","4":"hello","5":0,"6":1.0,"7":1.0}}
```

#### Testing
- Successfully ran `ninja -v -k 0 check`
- Verified that the output is the compact format that was desired.

### CODE REVIEW

**Comment by CodeChronos928** [comment]
If my PR is viewed as a favorable change, I'll get the failing unit tests updated.

**Comment by CodeChronos928** [comment]
@mrjerryjohns - I removed your "#if 0" block, as I believe I achieved your original goal.  Instead of comparing Json::Values though, I normalized the reference string to be in the format expected from the library.  This ensures the library is generating the expected format, and also lifts any white-space requirements in the reference strings.  I left your reference strings as-is since they are easier to read when examining a test failure or writing new tests.

**Review by turon — APPROVED** [review]
:+1: LGTM

**Review by arkq — APPROVED** [review]

**Review by mrjerryjohns — APPROVED** [review]

File: src/lib/support/jsontlv/TlvJson.cpp [hunk 1]
```diff
@@ -106,7 +106,8 @@ void InsertKeyValue(Json::Value & json, const KeyContext & keyContext, T val)
 
 std::string JsonToString(Json::Value & json)
```
[the remark this answers is not in the document]
Comment (CodeChronos928) [reply src/lib/support/jsontlv/TlvJson.cpp:107]: Noted, but I'd prefer to wait until someone expresses a need for the debug mode.  As you've approved the changes, I'm assuming you're ok with that.

File: src/lib/support/tests/TestTlvToJson.cpp [hunk 2]
```diff
@@ -54,55 +54,30 @@ CHIP_ERROR SetupReader()
...
+    // Normalize the reference string to the expected compact value.
+    Json::Reader reader;
+    Json::Value referenceValue;
+    reader.parse(referenceString, referenceValue);
+
+    Json::FastWriter writer;
+    writer.omitEndingLineFeed();
+    auto compactReferenceString = writer.write(referenceValue);
+
+    auto matches = (generatedStr == compactReferenceString);
```
[the remark this answers is not in the document]
Comment (CodeChronos928) [reply src/lib/support/tests/TestTlvToJson.cpp:68]: Thanks!

File: src/lib/support/jsontlv/TlvJson.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrjerryjohns) [reply src/lib/support/jsontlv/TlvJson.cpp:107]: Fair enough.

**Pull request merged**